### PR TITLE
Author Name Recipe

### PR DESCRIPTION
**Pull request by okbel**
## What does this PR do?
- New Recipe: Author Name Recipe: **Member Since**
  https://www.pivotaltracker.com/story/show/150178457

### Author Name Recipe: **Member Since** - Preview
screen shot 2017-08-21 at 12 15 26 pm

### CODE REVIEW

File: plugins/author-name/client/components/Tooltip.js [hunk 1]
```diff
@@ -0,0 +1,16 @@
...
+import styles from './Tooltip.css';
+import {t} from 'plugin-api/beta/client/services';
+import {Icon} from 'plugin-api/beta/client/components/ui';
+import moment from 'moment';
+
+export default ({className = '', memberSinceDate}) => (
+  <div className={cn(styles.tooltip, className)}>
+    <Icon name="date_range" className={styles.icon} />
+    <span className={styles.memberSince}>
+      {console.log(memberSinceDate)}
```
Comment (alinhle) [inline plugins/author-name/client/components/Tooltip.js]: Was this forgotten?

File: plugins/author-name/package.json [hunk 2]
```diff
@@ -0,0 +1,11 @@
+{
+  "name": "@coralproject/talk-plugin-author-name",
+  "version": "1.0.0",
```
Comment (alinhle) [inline plugins/author-name/package.json]: I prefer starting with `0.x.x` as the initial version number :-D

File: plugins/author-name/client/containers/AuthorName.js [hunk 3]
```diff
@@ -0,0 +1,46 @@
...
+
+  hideTooltip = () => {
+    this.setState({
+      tooltip: false
+    })
+  }
+
+  render() {
+    return <AuthorName 
+      {...this}
```
Comment (alinhle) [inline plugins/author-name/client/containers/AuthorName.js]: Avoid `{...this}` this just passes everything, even the `render` function. It's better to be explicit, so what is passed down is clearly defined: 

```js
<AuthorName 
  comment={this.props.comment}
  showTooltip={this.showTooltip}
  hideTooltip={this.hideTooltip}
  tooltipVisible={this.state.tooltip}
/>
```

File: plugins/author-name/client/components/AuthorName.js [hunk 4]
```diff
@@ -0,0 +1,20 @@
+import React from 'react';
+import {t} from 'plugin-api/beta/client/services';
+import Tooltip from './Tooltip';
+import styles from './AuthorName.css';
+
+export default ({props: {comment}, state: {tooltip}, showTooltip, hideTooltip}) => {
```
Comment (alinhle) [inline plugins/author-name/client/components/AuthorName.js]: See above/below/previous comment 😆

**Review by alinhle — CHANGES_REQUESTED** [review]
See comments

File: plugins/author-name/client/components/Tooltip.js (hunk 1 above)
Comment (okbel) [reply plugins/author-name/client/components/Tooltip.js]: Of course 😬

File: plugins/author-name/package.json (hunk 2 above)
Comment (okbel) [reply plugins/author-name/package.json]: 👍  I like it too

File: plugins/author-name/client/containers/AuthorName.js (hunk 3 above)
Comment (okbel) [reply plugins/author-name/client/containers/AuthorName.js]: Agreed.

File: plugins/author-name/client/containers/AuthorName.js [hunk 5]
```diff
@@ -0,0 +1,49 @@
...
+  hideTooltip = () => {
+    this.setState({
+      tooltip: false
+    })
+  }
+
+  render() {
+    return <AuthorName 
+      comment={this.props.comment}
+      tooltip={this.state.tooltip}
```
Comment (alinhle) [inline plugins/author-name/client/containers/AuthorName.js]: This should be `tooltipVisible={...`

File: plugins/author-name/client/components/Tooltip.js [hunk 6]
```diff
@@ -0,0 +1,15 @@
...
+import styles from './Tooltip.css';
+import {t} from 'plugin-api/beta/client/services';
+import {Icon} from 'plugin-api/beta/client/components/ui';
+import moment from 'moment';
+
+export default ({className = '', memberSinceDate}) => (
+  <div className={cn(styles.tooltip, className)}>
+    <Icon name="date_range" className={styles.icon} />
+    <span className={styles.memberSince}>
+      {t('talk-plugin-author-name.member_since')}: {moment(memberSinceDate).format('MMMM DD, YYYY')}
```
Comment (alinhle) [inline plugins/author-name/client/components/Tooltip.js]: image
According to `moment` documentation you must provide a format string as a second value: `moment(memberSinceData, dateFormat).format('MMMM DD, YYYY');`

**Review by alinhle — APPROVED** [review]

**Comment by alinhle** [comment]
I think we should add an entry in the README file. @okbel

**Pull request merged**